[torch op][xs] verbose error message for type mismatch in toList()

tgalkovskyi opened the pull request:
Summary:
Currently error message doesn't give you details on nature of mismatch:
  Output annotation element type and runtime tensor element type must match for tolist()

After update, error becomes actionable:
  RuntimeError: Output annotation element type and runtime tensor element type must match for tolist(): Long vs Int

Test Plan: existing unit tests

Reviewed By: iseeyuan

Differential Revision: D50082858

houseroad reviewed (approved):
Please ensure all the tests pass before landing.

tgalkovskyi committed:
[torch op][xs] verbose error message for type mismatch in toList() (#110872)

Summary:

Currently error message doesn't give you details on nature of mismatch:
  Output annotation element type and runtime tensor element type must match for tolist()

After update, error becomes actionable:
  RuntimeError: Output annotation element type and runtime tensor element type must match for tolist(): Long vs Int

Test Plan: existing unit tests

Reviewed By: iseeyuan

Differential Revision: D50082858
